app/ModelFunctions/LogFunctions.php

```diff
@@ -37,7 +38,7 @@ private function interpolate(string $message, array $context = []): string
 	/**
 	 * Implements log so that AbstractLogger works.
 	 */
-	public function log($loglevel, $message, $context = [])
+	public function log($log, Stringable|string $message, array $context = []): void
```

Comment:
> Method `App\ModelFunctions\LogFunctions::log()` is not compatible with method `Psr\Log\AbstractLogger::log()`. intelephense(1038)

Reply:
```php
<?php

namespace Psr\Log;

/**
 * Describes a logger instance.
 *
 * The message MUST be a string or object implementing __toString().
 *
 * The message MAY contain placeholders in the form: {foo} where foo
 * will be replaced by the context data in key "foo".
 *
 * The context array can contain arbitrary data. The only assumption that
 * can be made by implementors is that if an Exception instance is given
 * to produce a stack trace, it MUST be in a key named "exception".
 *
 * See https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md
 * for the full interface specification.
 */
interface LoggerInterface
{
...
    /**
     * Logs with an arbitrary level.
     *
     * @param mixed   $level
     * @param string|\Stringable $message
     * @param mixed[] $context
     *
     * @return void
     *
     * @throws \Psr\Log\InvalidArgumentException
     */
    public function log($level, string|\Stringable $message, array $context = []): void;
}
```

Reply:
> Method `App\ModelFunctions\LogFunctions::log()` is not compatible with method `Psr\Log\AbstractLogger::log()`.

This doesn't answer the question.

The signature of the parent class is

```php
 public function log($level, string|\Stringable $message, array $context = []): void
```

The previous signature of our method was

```php
 public function log($loglevel, string|\Stringable $message, array $context = []): void
```

and now you changed it to

```php
 public function log($log, string|\Stringable $message, array $context = []): void
```

The parameters name in the parent class is just `$level`. For our class it has been `$loglevel`. OK, that has been an inconsistency, but this no explanation for changing it to `$log`. First of all, it does not solve the inconsistency (if this has been the goal) and it as I said in the first comment, `$log` is much less telling than either `$loglevel` or `$level` because it can easily be confused with the actual log message.

Reply:
Fixed [9f06381](https://github.com/LycheeOrg/Lychee/pull/1469/commits/9f063810edba917a0fa1b989812eb5616b21f4d8)